packages/cascara/src/hooks/useMediaQuery.js

```diff
@@ -3,21 +3,28 @@
 import { useEffect, useState } from 'react';
 
 const useMediaQuery = (query, ifTrueFunc, ifFalseFunc) => {
-  const mediaQuery = window.matchMedia(query);
+  const [mediaQuery, setMediaQuery] = useState();
+
   // We need to set state here so that the hook will return the new value
-  const [match, setMatch] = useState(Boolean(mediaQuery.matches));
+  const [match, setMatch] = useState();
+
+  useEffect(() => {
+    const mediaQuery = window.matchMedia(query);
+
+    setMediaQuery(mediaQuery);
+  }, [query]);
```

Comment:
@lordpixel is there a reason not to just move the logic from this useEffect inside of the useEffect that is directly below on line 17? It all needs to be done at the same time, and they will not be done separately. If this needs the useEffect, I think it makes sense to have it all inside the same effect hook.

Reply:
@brianespinosa thanks for your help on this one. 👍